src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/EligibilityAndSuitabilityService.kt

```diff
@@ -102,7 +105,13 @@ class EligibilityAndSuitabilityService(
         criterionMet,
         answer.answers,
       )
-      assessmentEntity.changeStatus(assessmentLifecycleService.eligibilityAnswerSubmitted(currentAssessment))
+      val event = when (currentAssessment.calculateAggregateEligibilityStatus()) {
+        ELIGIBLE -> AssessmentLifecycleEvent.CompleteEligibilityAndSuitability
+        INELIGIBLE -> AssessmentLifecycleEvent.FailEligibilityAndSuitability
+        IN_PROGRESS -> AssessmentLifecycleEvent.StartEligibilityAndSuitability
+        NOT_STARTED -> error("Assessment: ${assessmentEntity.id} is in an unexpected state: ${assessmentEntity.status}")
```

Comment:
It's possible that `eligibilitystatus` should be an attribute of a new `ElgibilityAnswerSupplied` event and then the state transition could determine where to go in the statemachine:
```
  state<AssessmentState.ElgibilityAnswerSupplied> {
    on<AssessmentLifecycleEvent.CompleteEligibilityAndSuitability> {
      val newStage = when(it.eligiblityStatus) {
        ELIGIBLE -> AssessmentLifecycleEvent.CompleteEligibilityAndSuitability
        INELIGIBLE -> AssessmentLifecycleEvent.FailEligibilityAndSuitability
        IN_PROGRESS -> AssessmentLifecycleEvent.StartEligibilityAndSuitability
        NOT_STARTED -> error("Assessment: ${assessmentEntity.id} is in an 
      }
      transitionTo(newStage)
    }
```

There's lots of different states that this event could be applied to so we'd need to look at reusing this kind of logic and applying it to multiple stages